fendermint/app/src/app.rs

```diff
@@ -666,27 +688,43 @@ where
             "process proposal"
         );
         let txs: Vec<_> = request.txs.into_iter().map(|tx| tx.to_vec()).collect();
+        let size_txs = txs.iter().map(|tx| tx.len()).sum::<usize>();
         let num_txs = txs.len();
 
-        let accept = self
+        let process_result = self
             .interpreter
             .process(self.chain_env.clone(), txs)
-            .await
-            .context("failed to process proposal")?;
```

Comment:
Dug into this further:

- Returning an error from the `Application` will induce a panic and cause the Tower Service to crash. 
- This approach is sensible if (a) Tower is configured to restart the service upon crash, AND (b) we're in the middle of some state update (e.g. after begin_block and before end_block). In that case, we want to reset transient state and start over.
- However, (b) is not true for process_proposal -- this is a stateless method. So returning an error here is unnecessary and problematic.

Conclusion:

- Thanks to your diff, we discovered a deeper issue! 🌟 
- But because this is a delicate part of the system, I prefer to extract this diff out to its own changeset and changelog entry.
- Behaviourally speaking, these changes seem reasonable, but I have some hesitations on introducing new types to signal a binary outcome (idiomatic way is to use Result or Option, either flattened or nested).

Next steps:

- Let's back out these changes and forego setting a reason for now.
- Open a separate PR with your changes that also prevents `process_proposal` from propagating an error.